### PR TITLE
Feature random seed

### PR DESCRIPTION
**Pull request by mducducd**
Add a Seed utility class to simplify and unify random seed management across PyTorch, NumPy, and Python’s random. It provides methods to set global seeds and initialize DataLoader workers reproducibly. Now available in advanced_config in config.yaml.

### CODE REVIEW

File: src/stamp/__main__.py [hunk 1]
```diff
@@ -52,9 +52,13 @@ def _run_cli(args: argparse.Namespace) -> None:
 
     if config.advanced_config is None:
         config.advanced_config = AdvancedConfig(
-            model_params=ModelParams(vit=VitModelParams(), mlp=MlpModelParams())
```
Comment (EzicStar) [inline src/stamp/__main__.py:55]: Is there a reason to remove the initialization of the default values for the selected models?

File: src/stamp/modeling/data.py [hunk 2]
```diff
@@ -101,10 +101,8 @@ def tile_bag_dataloader(
                 shuffle=shuffle,
                 num_workers=num_workers,
                 collate_fn=_collate_to_tuple,
-                worker_init_fn=Seed.get_loader_worker_init()
-                if Seed._is_set()
-                else None,
-                generator=Seed.get_torch_generator() if Seed._is_set() else None,
+                worker_init_fn=Seed.get_loader_worker_init(),
```
Comment (EzicStar) [inline src/stamp/modeling/data.py:104]: Won´t this cause an error if a seed is not set?

**Review by EzicStar — CHANGES_REQUESTED** [review]
I have added some comments on changes that could introduce potential errors. Feel free to ask and/or disagree on them

File: src/stamp/__main__.py (hunk 1 above)
Comment (mducducd) [reply src/stamp/__main__.py:55]: Because it's possible to set it to None when we only get the Seed (None or int) from here.
However, we can initialize the full default models and values from here; therefore, checking for None at "crossval" and "training" is no longer needed. What do you think?

Comment (EzicStar) [reply src/stamp/__main__.py:55]: Nice! I like the idea

**Review by EzicStar — APPROVED** [review]
LGTM!
